pkg/chart/chart.go

```diff
@@ -364,11 +346,19 @@ func (c *Client) GetChart(details *Details, netClient kube.HTTPClient, requireV1
 
 // RegistrySecretsPerDomain checks the app repo and available secrets
 // to return the secret names per registry domain.
-//
-// These are actually calculated during InitNetClient when we already have a
-// k8s client with the user token.
-func (c *Client) RegistrySecretsPerDomain() map[string]string {
-	return c.registrySecretsPerDomain
+func RegistrySecretsPerDomain(appRepo *appRepov1.AppRepository, cluster string, userAuthToken string, authHandler kube.AuthHandler) (map[string]string, error) {
+	registrySecretsPerDomain, err := getRegistrySecretsPerDomain(
```

Comment:
I don't think there's any need for the indirection now, you could just have the implementation of (the private) `getRegistrySecretsPerDomain` here now, I think? (what's here is essentially now `return getRegistrySecretsPerDomain(...)`)